libraries/AP_Terrain/TerrainIO.cpp

```diff
@@ -212,19 +212,8 @@ void AP_Terrain::seek_offset(void)
 {
     struct grid_block &block = disk_block.block;
     // work out how many longitude blocks there are at this latitude
-    Location loc1, loc2;
-    loc1.lat = block.lat_degrees*10*1000*1000L;
-    loc1.lng = block.lon_degrees*10*1000*1000L;
-    loc2.lat = block.lat_degrees*10*1000*1000L;
-    loc2.lng = (block.lon_degrees+1)*10*1000*1000L;
-
-    // shift another two blocks east to ensure room is available
-    loc2.offset(0, 2*grid_spacing*TERRAIN_GRID_BLOCK_SIZE_Y);
-    const Vector2f offset = loc1.get_distance_NE(loc2);
-    uint16_t east_blocks = offset.y / (grid_spacing*TERRAIN_GRID_BLOCK_SIZE_Y);
```

Comment:
Stride was too small because blocks are overlapping and spaced by TERRAIN_GRID_BLOCK_SPACING_Y
https://github.com/avrem/ardupilot/commit/da24fa4bed864e45eadb6fa7c921186703a2f74d
With this fix, variable-per-degree stride works OK.

Reply:
Thanks! I'll fix it up tomorrow

Reply:
fixed, thanks for your help